lib/features/network_connection/presentation/network_connection_controller.dart

```diff
@@ -4,73 +4,109 @@ import 'package:connectivity_plus/connectivity_plus.dart';
-    final currentConnectionResult = await _connectivity.checkConnectivity();
-    log('NetworkConnectionController::onReady():_getCurrentNetworkConnectionState: $currentConnectionResult');
-    _setNetworkConnectivityState(currentConnectionResult);
+    final listConnectionResult = await Future.wait([
+      _connectivity.checkConnectivity(),
+      if (PlatformInfo.isMobile)
+        _internetConnectionChecker.connectionStatus,
+    ]);
+    log('NetworkConnectionController::_getCurrentNetworkConnectionState():listConnectionResult: $listConnectionResult');
+
+    if (listConnectionResult[0] is ConnectivityResult) {
+      _setNetworkConnectivityState(listConnectionResult[0] as ConnectivityResult);
+    }
+
+    if (PlatformInfo.isMobile && listConnectionResult[1] is ConnectivityResult) {
+      _setInternetConnectivityStatus(listConnectionResult[1] as InternetConnectionStatus);
+    }
+
     _handleNetworkConnectionState();
   }
 
   void _listenNetworkConnectionChanged() {
     subscription = _connectivity.onConnectivityChanged.listen(
       (result) {
-        log('NetworkConnectionController::_listenNetworkConnectionChanged():onConnectivityChanged: $result');
+        log('NetworkConnectionController::_listenNetworkConnectionChanged()::onConnectivityChanged: $result');
         _setNetworkConnectivityState(result);
         _handleNetworkConnectionState();
       },
       onError: (error, stackTrace) {
-        logError('NetworkConnectionController::_listenNetworkConnectionChanged():error: $error');
-        logError('NetworkConnectionController::_listenNetworkConnectionChanged():stackTrace: $stackTrace');
+        logError('NetworkConnectionController::_listenNetworkConnectionChanged()::onConnectivityChanged:error: $error | stackTrace: $stackTrace');
       }
     );
+
+    if (PlatformInfo.isMobile) {
+      internetSubscription = _internetConnectionChecker.onStatusChange.listen(
+        (status) {
+          log('NetworkConnectionController::_listenNetworkConnectionChanged()::onStatusChange: $status');
+          _setInternetConnectivityStatus(status);
+        },
+        onError: (error, stackTrace) {
+          logError('NetworkConnectionController::_listenNetworkConnectionChanged()::onStatusChange:error: $error | stackTrace: $stackTrace');
+        }
+      );
+    }
   }
 
   void _setNetworkConnectivityState(ConnectivityResult newConnectivityResult) {
     connectivityResult.value = newConnectivityResult;
   }
 
+  void _setInternetConnectivityStatus(InternetConnectionStatus newStatus) {
+    internetConnectionStatus.value = newStatus;
+  }
+
   bool isNetworkConnectionAvailable() {
-    return connectivityResult.value != ConnectivityResult.none;
+    if (PlatformInfo.isWeb) {
+      return connectivityResult.value != ConnectivityResult.none;
+    } else {
+      return connectivityResult.value != ConnectivityResult.none &&
+        internetConnectionStatus.value == InternetConnectionStatus.connected;
```

Comment:
IMO, `internetConnectionStatus.value == InternetConnectionStatus.connected;` is also a network connected

Reply:
I do not understand what you mean?